ops/testing.py

```diff
@@ -1141,6 +1141,67 @@ def reset_planned_units(self):
         """
         self._backend._planned_units = None
 
+    def add_network(self, binding_name: str, address: str, *,
+                    relation_id: Optional[int] = None,
+                    cidr: Optional[str] = None,
+                    interface: str = 'eth0',
+                    ingress_addresses: Optional[List[str]] = None,
+                    egress_subnets: Optional[List[str]] = None,
+                    ):
+        """Add simulated network data for the given binding.
+
+        Calling this multiple times with the same (binding_name, relation_id)
+        combination will replace the associated network data.
+
+        This method only supports the common case of adding a single interface
+        and bind address. If you need support for multiple bind addresses, you
+        can patch ops.testing._TestingModelBackend.network_get and return the
+        full network-get data via the patched method.
+
+        Args:
+            binding_name: Name of binding (endpoint) to add network data for.
+                Use "" (empty string) to set the default binding.
+            address: Binding's IP address, for example "10.0.0.10".
```

Comment:
Can this be either an IPv4 or IPv6 address?

Reply:
Yep, good point. Updated.